vizro-ai/src/vizro_ai/_vizro_ai.py

```diff
@@ -48,120 +38,41 @@ def __init__(self, model: Optional[Union[ChatOpenAI, str]] = None):
-        return self.components_instances[component_class]
-
-    def _run_plot_tasks(
-        self,
-        df: pd.DataFrame,
-        user_input: str,
-        max_debug_retry: int = 3,
-        explain: bool = False,
-        chart_name: Optional[str] = None,
-    ) -> PlotOutputs:
-        """Task execution."""
-        chart_type_pipeline = self.pipeline_manager.chart_type_pipeline
-        chart_type = chart_type_pipeline.run(initial_args={"chain_input": user_input, "df": df})
-
-        # TODO update to loop through charts for multiple charts creation
-        if chart_name is None:
-            chart_name = "custom_chart"
-        plot_pipeline = self.pipeline_manager.plot_pipeline
-        custom_chart_code = plot_pipeline.run(
-            initial_args={"chain_input": user_input, "df": df, "chart_type": chart_type, "chart_name": chart_name}
-        )
-
-        # TODO add debug in pipeline after getting _debug_helper logic in component
-        fix_func = self._lazy_get_component(GetDebugger).run
-        validated_code_dict = _debug_helper(
-            code_string=custom_chart_code, max_debug_retry=max_debug_retry, fix_chain=fix_func, df=df
-        )
-
-        pass_validation = validated_code_dict.get("debug_status")
-        code_string = validated_code_dict.get("code_string")
-
-        if not pass_validation:
-            raise DebugFailure(
-                "Chart creation failed. Retry debugging has reached maximum limit. Try to rephrase the prompt, "
-                "or try to select a different model. Fallout response is provided: \n\n" + code_string
-            )
-
-        fig_object = _exec_code_and_retrieve_fig(code=code_string, local_args={"df": df}, is_notebook_env=_is_jupyter())
-        if explain:
-            business_insights, code_explanation = self._lazy_get_component(GetCodeExplanation).run(
-                chain_input=user_input, code_snippet=code_string
-            )
-
-            return PlotOutputs(
-                code=code_string,
-                figure=fig_object,
-                business_insights=business_insights,
-                code_explanation=code_explanation,
-            )
-
-        return PlotOutputs(code=code_string, figure=fig_object)
-
-    def _get_chart_code(self, df: pd.DataFrame, user_input: str) -> str:
-        """Get Chart code of vizro via english descriptions, English to chart translation.
-
-        Can be used in integration with other application if only code snippet return is required.
-
-        Args:
-            df: The dataframe to be analyzed
-            user_input: User questions or descriptions of the desired visual
-
-        """
-        # TODO retained for some chat application integration, need deprecation handling
-        return self._run_plot_tasks(df, user_input, explain=False).code
 
     def plot(
         self,
         df: pd.DataFrame,
         user_input: str,
-        explain: bool = False,
```

Comment:
I like the idea of removing `explain` from here, and using `return_elements` to cover it (which is much more streamlined). Although this is a small breaking change, we didn't provide any deprecation warning before fully remove it.

Just to make users' life easier, shall we maybe try improve the error message in case user uses `explain`? and provide a migration guide (or as simple as a suggested code block), so it's clear how to get explanations with new API. Something like

```
def deprecated_param_check(func):
    @wraps(func)
    def wrapper(*args, **kwargs):
        if 'explain' in kwargs:
            raise TypeError(
                "VizroAI.plot() no longer supports the 'explain' parameter. "
                "This parameter has been removed in the latest version. "
                "If you need explanations, do `return_elements=True` and `res.chart_insights` and `res.code_explanation`"
            )
        return func(*args, **kwargs)
    return wrapper
    
    
class VizroAI:
    @deprecated_param_check
    def plot()
```

Current error message is
```
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
Cell In[8], line 1
----> 1 vizro_ai.plot(df, "describe the composition of gdp in continent, and add horizontal line for avg gdp", explain=True)

TypeError: VizroAI.plot() got an unexpected keyword argument 'explain'
```

Reply:
Good idea, I like it!

Technically we could also implement this for the returned object, but that seems a little weird on the `ChartPlanStatic` no?